src/main/java/com/google/javascript/cl2dts/DeclarationGenerator.java

```diff
@@ -490,6 +496,8 @@ public Void caseStringType() {
         public Void caseObjectType(ObjectType type) {
           if (type.isRecordType()) {
             visitRecordType((RecordType) type);
+          } else if (type.isDict()) {
```

Comment:
But in that case we fail to produce a member for `foo` here, right? I'll file an issue.